src/libxrpl/protocol/Permissions.cpp

```diff
@@ -118,15 +132,52 @@ Permission::getGranularTxType(GranularPermissionType const& gpType) const
 }
 
 bool
-Permission::isDelegatable(std::uint32_t const& permissionValue) const
+Permission::isDelegatable(
+    std::uint32_t const& permissionValue,
+    Rules const& rules) const
 {
     auto const granularPermission =
         getGranularName(static_cast<GranularPermissionType>(permissionValue));
     if (granularPermission)
         // granular permissions are always allowed to be delegated
         return true;
 
-    auto const it = delegatableTx_.find(permissionValue - 1);
+    auto const txType = permissionToTxType(permissionValue);
+    auto const it = delegatableTx_.find(txType);
+
+    if (rules.enabled(fixDelegateV1_1))
+    {
+        if (it == delegatableTx_.end())
+            return false;
+
+        auto const txFeaturesIt = txFeaturesMap_.find(txType);
+        XRPL_ASSERT(
+            txFeaturesIt != txFeaturesMap_.end(),
+            "ripple::Permissions::isDelegatable : tx does not exist in "
+            "txFeaturesMap_");
+
+        // fixDelegateV1_1: Delegation is not allowed if any required amendment
+        // for the transaction is not enabled.
+        if (std::any_of(
+                txFeaturesIt->second.begin(),
+                txFeaturesIt->second.end(),
+                [&rules](auto const& feature) {
+                    if (feature == featureNonFungibleTokensV1 ||
+                        feature == fixNFTokenNegOffer ||
+                        feature == fixNFTokenDirV1)
+                    {
+                        // The functionality of the featureNonFungibleTokensV1_1
+                        // amendment is precisely the functionality of the three
+                        // amendments above.
+                        return !rules.enabled(featureNonFungibleTokensV1_1) &&
+                            !rules.enabled(feature);
+                    }
+
+                    return !rules.enabled(feature);
+                }))
+            return false;
```

Comment:
I think checking for the negative here is making this unnecessarily complicated, and causing confusion.

```suggestion
        // fixDelegateV1_1: Delegation is only allowed if all required amendment(s)
        // for the transaction are enabled.
        if (!std::all_of(
                txFeaturesIt->second.begin(),
                txFeaturesIt->second.end(),
                [&rules](auto const& feature) {
                    return rules.enabled(feature);
                }))
            return false;
```